### ISSUE TITLE
A setting to disable the React error overlay

### ISSUE THREAD

**Issue by LoganDark** [parcel-bundler/parcel#9738]
The Parcel error overlay freezes our app for about 10 seconds every time there's an error, which is a constant annoyance and significantly slows down development. It looks like there are no settings to control it whatsoever.

I would like to be able to turn the error overlay off. I don't need the app to be parsing source maps for 10 seconds every time an error occurs - DevTools does that for me.

**Comment by hunxjunedo** [comment]
This is an enhancement and new feature, which requires discussion with the maintainers: @devongovett @mischnic

**Comment by LoganDark** [comment]
> This is an enhancement and new feature

Perfectly fine by me. I did click the new feature button, but I guess since it doesn't assign any labels that wasn't obvious.

**Comment by LoganDark** [comment]
![image](https://github.com/parcel-bundler/parcel/assets/4723091/93c3833a-f19b-4b2c-a2bb-aae4a428794f)

**Comment by hunxjunedo** [comment]
Damn that looks like a real mess, but still starting to work on this without consulting isn't a good idea.

**Comment by LoganDark** [comment]
> Damn that looks like a real mess, but still starting to work on this without consulting isn't a good idea.

Yeah, don't worry, I'm fine waiting for the maintainers' input. That wasn't intended as a nag, just to show how bad the issue is (it slows testing down by a factor of 2-3x just because I have to sit and wait between every click). A lot of these errors are unrelated to what I'm doing since other people are working on large refactors, so that's why I haven't just fixed them myself.

**Comment by LoganDark** [comment]
it's possible to use the debugger to get the overlay to STFU. set a conditional breakpoint that overwrites `require`:

![image](https://github.com/parcel-bundler/parcel/assets/4723091/3ad26bcc-f404-43e6-9f01-d04b7ccc3767)

`void (require = ((require) => (module) => (console.log('require(%o)', module), module) === 'e88bec6ec2bff47' ? { setEditorHandler: () => {}, startReportingRuntimeErrors: () => {}, dismissRuntimeErrors: () => {} } : require(module))(require))`

if it doesn't fire, replace `e88bec6ec2bff47` with whatever is the *second* log to the console here: ![image](https://github.com/parcel-bundler/parcel/assets/4723091/d0ae8aea-75af-4aad-8bb9-bc8b9d95415f)

no more 15-second freezes for everything (even errors caught by error boundaries) !!!!!

**Comment by LoganDark** [comment]
@mischnic do you have any thoughts on this? would you accept a PR? we're still impacted by this issue

**Comment by kai-dorschner-twinsity** [comment]
Bump

**Comment by LoganDark** [comment]
stfu stale bot

**Comment by devongovett** [comment]
Small update: in the next release we have a refactor to the error overlay to modernize it and improve performance. The expensive source mapping stuff will be done in native code in the dev server instead of in the browser so it should be much faster. https://github.com/parcel-bundler/parcel/pull/10082

Open to adding a flag to disable it entirely as well but hopefully that helps with the underlying reason to do so in the first place.

**Comment by LoganDark** [comment]
that refactor sounds very promising !!

would you be able to toss in a flag just in case? for me it would be nice to know the option exists, even if maybe (hopefully) I won't have to use it

**Comment by cdiaz-atlassian** [comment]
This overlay is actually very disruptive and the user should be allowed to opt-out of that "feature". The last thing you need while debugging is an overlay that blocks the view and that you need to manually close to be able to see what you're working on. Everything is being logged to the console any way and if I want to see errors I can look there.

**Comment by cdiaz-atlassian** [comment]
@LoganDark I took a more "permanent" temporary approach until this is fixed. I used [patch-package](https://www.npmjs.com/package/patch-package) to patch @parcel/error-overlay this might also work for you in the meantime. 

```diff
diff --git a/node_modules/@parcel/error-overlay/lib/index.js b/node_modules/@parcel/error-overlay/lib/index.js
index c7abbdc..26999f8 100644
--- a/node_modules/@parcel/error-overlay/lib/index.js
+++ b/node_modules/@parcel/error-overlay/lib/index.js
@@ -1902,6 +1902,7 @@ function $da9882e673ac146b$export$cda2c88a41631c16(options) {
     $da9882e673ac146b$var$stopListeningToRuntimeErrors = (0, $6d40ebe8356580e0$export$38ec23daa6e8dcdf)($da9882e673ac146b$var$handleRuntimeError(options));
 }
 const $da9882e673ac146b$var$handleRuntimeError = (options)=>(errorRecord)=>{
+    return
         try {
             if (typeof options.onError === 'function') options.onError.call(null);
         } finally{

```

**Comment by wojciechb** [comment]
As much as I like the overlay for errors, it shouldn't be triggered fro every single react warning.

Extremely disruptive, given no obvious or even clean way to disable it. Almost forced me out of parcel environment, and I wasted quite a while to hack it out.

**Comment by tehpsalmist** [comment]
Did that flag ever make it in?

**Comment by mcurros2** [comment]
OK, so while waiting for the setting to disable the overlay, you can disable it like this:

In the package.json
`
"alias": {
  "@parcel/error-overlay": "./src/no-error-overlay.js"
}
`

The no-error-overlay.js

```
// Overrides to disable error overlay

export function setEditorHandler() { }

export function reportRuntimeError() {}

export function startReportingRuntimeErrors() {}

export function dismissRuntimeErrors() {}

export function stopReportingRuntimeErrors() {}
```

In our case we prefer to lookup for errors in the console. The overlay is intrusive and takes focus out from what we are trying to solve. I hope you add the setting to disable it.

**Comment by emi-wheee** [comment]
Hi, people who showed up here via search engine!

If you happen to still be on an older Parcel from before they forked into `@parcel/error-overlay` (we're on parcel 2.13.3), aliasing `"react-error-overlay"` to @mcurros2's stub above works without any other changes.

**Comment by Tinkercode** [comment]
Do you think this can still be added as a flag? I want to avoid a workaround.

**Comment by jorge-fernandes** [comment]
Just updated parcel and got this... i find it a bit unbelievable a feature that can't be turned off was peer reviewed and release. Can you pls add a flag to disable it to improve DX?

**Comment by LoganDark** [comment]
Not sure that this issue is fixed yet.
